### PR TITLE
(test) O3-2108: Add medication e2e tests

### PR DESCRIPTION
**Pull request by RandilaP**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
Goal of this PR is to add medication e2e tests. This e2e test covered the medication functionality with add, modify, discontinue and reorder medication e2e tests. It has some issues with loading the tables after save the form. 

## Related Issue

https://issues.openmrs.org/browse/O3-2108

### CODE REVIEW

**Comment by RandilaP** [comment]
@jayasanka-sack @anjula-sack @Piumal1999 Could you please review this

File: e2e/commands/index.ts [hunk 1]
```diff
@@ -1 +1,2 @@
 export * from './patient-operations';
+export * from './visitOperations';
```
Comment (anjula-sack) [inline e2e/commands/index.ts:2]: Use kebab case for file naming @RandilaP 

```suggestion
export * from './visitOperations';

```

File: e2e/pages/index.ts [hunk 2]
```diff
@@ -1 +1,2 @@
 export * from './allergies-page';
+export * from './medications-page';
```
Comment (anjula-sack) [inline e2e/pages/index.ts:2]: Add EOL

```suggestion
export * from './medications-page';

```

File: e2e/specs/medications.spec.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        await expect(medicationPage.activeMedicationTable().getByText('DOSE 2 tablet — oral — Twice daily')).toBeVisible();
+        await expect(medicationPage.activeMedicationTable().getByText(' for 30 days — REFILLS 6')).toBeVisible();
+        await expect(medicationPage.activeMedicationTable().getByText('QUANTITY 60 Tablet')).toBeVisible();
+    });
+});
+
+test.afterEach(async ({ api }) => {
+    await endVisit(api, patient.uuid);
+    await deletePatient(api, patient.uuid);
+  });
```
Comment (anjula-sack) [inline e2e/specs/medications.spec.ts:116]:
```suggestion
  });
  
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: e2e/specs/medications.spec.ts (hunk 3 above)
Comment (RandilaP) [reply e2e/specs/medications.spec.ts]: @anjula-sack Done

File: e2e/pages/index.ts (hunk 2 above)
Comment (RandilaP) [reply e2e/pages/index.ts]: @anjula-sack Done

File: e2e/commands/index.ts (hunk 1 above)
Comment (RandilaP) [reply e2e/commands/index.ts]: @anjula-sack  Done

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: e2e/commands/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (RandilaP) [reply e2e/commands/index.ts]: @anjula-sack Done

File: e2e/specs/medications.spec.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
```
[the remark this answers is not in the document]
Comment (RandilaP) [reply e2e/specs/medications.spec.ts:116]: @anjula-sack Done

File: e2e/specs/medications.spec.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      .getByPlaceholder('Additional dosing instructions (e.g. "Take after eating")')
+      .fill('Test comment');
+    await medicationPage.page.getByPlaceholder('mm/dd/yyyy').fill('07/10/2023');
+    await medicationPage.page.getByPlaceholder('mm/dd/yyyy').press('Tab');
+    await medicationPage.page.getByLabel('Duration', { exact: true }).fill('30');
+    await medicationPage.page.getByLabel('Quantity to dispense').fill('30');
+    await medicationPage.page.getByLabel('Prescription refills').fill('3');
+    await medicationPage.page.getByPlaceholder('e.g. "Hypertension"').fill('Mental status change');
+    await medicationPage.page.getByRole('button', { name: 'Save order' }).click();
+    await medicationPage.page.getByRole('button', { name: 'Sign and close' }).click();
```
Comment (anjula-sack) [inline e2e/specs/medications.spec.ts:55]: This is a separate step

File: e2e/specs/medications.spec.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      .getByRole('listitem')
+      .filter({ hasText: 'ModifyAspirin 162.5mg — 162.5mg — TabletDOSE 1 Tablet — Oral — Once daily — REFI' })
+      .click();
+    await medicationPage.page.getByPlaceholder('Dose').fill('2');
+    await medicationPage.page.getByPlaceholder('Frequency').click();
+    await medicationPage.page.getByText('Twice daily').click();
+    await medicationPage.page.getByLabel('Quantity to dispense').fill('60');
+    await medicationPage.page.getByLabel('Prescription refills').fill('6');
+    await medicationPage.page.getByRole('button', { name: 'Save order' }).click();
+    await medicationPage.page.getByRole('button', { name: 'Sign and close' }).click();
```
Comment (anjula-sack) [inline e2e/specs/medications.spec.ts:80]: Same here

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — CHANGES_REQUESTED** [review]
@RandilaP can you resolve the conflicts?

**Comment by RandilaP** [comment]
@anjula-sack What should we do with this test

**Comment by anjula-sack** [comment]
> @anjula-sack What should we do with this test

Can you check again the report?  The form is not saving @RandilaP

**Comment by RandilaP** [comment]
@anjula-sack The orders aren't saving because of some errors. I tried it manually and it didn't work

**Comment by brandones** [comment]
Closing as stale.

**Pull request closed without merging**